### PR TITLE
Fixes #251 / Add missing types to AD plugin

### LINKED ISSUE #251: ActiveDirectory plugin types are incomplete
The type definitions for the AD types are incomplete/incorrect

[Description of the issue. Affected functionality / feature.]

### Steps to Reproduce

n/a

**Preconditions:**

n/a

**Expected behavior:**

```ts
export declare class AD_Unknown {
  public readonly id: string
  public readonly distinguishedName: string
  public readonly allAttributes: any[]
  public readonly gUID: string
  public getArrayAttribute (attribName: string): string[]
  public getAttributeValueBytes (attribName: string): any[]
  public removeAttribute (attibName: string): void
  public destroy (param0: boolean): void
  public rename (name: string): void
  public setAttribute (attribName: string, newValue: any): void
  public getAttribute (attribName: string): string
  public addAttribute (attribName: string, newValue: any): void
}

export declare class AD_User extends AD_Unknown {
  public readonly sID: string
  public readonly memberof: AD_Group[]
  public readonly userPrincipalName: string
  public readonly accountName: string
  public readonly enabled: boolean
  public setPassword (password: string): void
  public setChangePasswordAtNextLogon (value: string): boolean
}

export declare class AD_Group extends AD_Unknown {
  public readonly userGroups: any[]
  public readonly computers: any[]
  public readonly containers: any[]
  public readonly organizationalUnits: any[]
  public readonly users: any[]
  public createUser (accountName: string, domainName: string, displayName: string)
}

export declare class AD_UserGroup extends AD_Unknown {
  public readonly sID: string
  public readonly computerMembers: any[]
  public readonly groupMembers: AD_UserGroup[]
  public readonly userMembers: AD_User[]
  public readonly memberOf: AD_UserGroup[]
}
```

**Actual behavior:**

```ts
declare interface AD_Group {
	id: string;
}

/**
 * Unknown type of object
 */
declare interface AD_Unknown {
	id: string;
}

/**
 * User
 */
declare interface AD_User {
	id: string;
}

/**
 * UserGroup
 */
declare interface AD_UserGroup {
	id: string;
}
```

**Reproduces how often:**

100%

**Component/s:**

node_modules/@types/o11n-plugin-activedirectory/index.d.ts

**Affects Build/s:**

all <2.37.0

#### Client

-   Build Tools for VMware Aria Version: 2.37.0
-   Visual Studio Code Version: n/a
-   OS Version: n/a

#### Server

-   vRealize Automation Version: n/a
-   vRealize Orchestrator Version: n/a
-   vRealize Operations Version: n/a
-   vRealize Log Insight Version: n/a

### Failure Logs
n/a

### PR DESCRIPTION
**Pull request by unbreakabl3**
### Description

Add missing type to AD plugin to close issue #251

### Checklist

- [x] I have added relevant error handling and logging messages to help troubleshooting
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation, relevant usage information (if applicable)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [ ] I have tested against live environment, if applicable
- [ ] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)
- [ ] I have my changes rebased and squashed to the minimal number of relevant commits. **Notice: don't squash all commits**
- [ ] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### CODE REVIEW

**Comment by VenelinBakalov** [comment]
@unbreakabl3 thanks for taking the time to update this! I left just 3 small comments, if you could check those

**Comment by unbreakabl3** [comment]
> @unbreakabl3 thanks for taking the time to update this! I left just 3 small comments, if you could check those

Hi, @VenelinBakalov.
Sure. But not sure I can see where the comments are :)

**Comment by VenelinBakalov** [comment]
@unbreakabl3 hmm good point, I see the review was not posted. I added it now. You might also need to pull from latest master to resolve a conflict

**Comment by VenelinBakalov** [comment]
@unbreakabl3 we needed to trigger a release, if you want wait 1 hour for the changes so that the release does not create another merge conflict on your PR after the fixes

**Comment by VenelinBakalov** [comment]
> > @unbreakabl3 we needed to trigger a release, if you want wait 1 hour for the changes so that the release does not create another merge conflict on your PR after the fixes
> 
> I will not do anything and wait. Take your time.

Release is ready, thanks!

File: vro-types/o11n-plugin-activedirectory/index.d.ts [hunk 1]
```diff
@@ -490,21 +563,183 @@ declare interface AD_OrganizationalUnit {
...
  * User
  */
 declare interface AD_User {
-	id: string;
+	readonly id: string;
+	readonly enabled: boolean;
+	readonly userPrincipalName: string;
+	readonly sID: string;
+	readonly accountName: string;
+	readonly memberof: AD_Group[];
```
[the remark this answers is not in the document]
Comment (unbreakabl3) [reply vro-types/o11n-plugin-activedirectory/index.d.ts:621]: Fixed

File: vro-types/o11n-plugin-activedirectory/index.d.ts [hunk 2]
```diff
@@ -490,21 +563,183 @@ declare interface AD_OrganizationalUnit {
...
+
+	/**
+	 * Get an AD attribute value as byte array.
+	 */
+	addAttribute(attribName: string): object[];
+
+	/**
+	 * Get an AD attribute for an array of values.
+	 */
+	addAttribute(attribName: string): string[];
```
[the remark this answers is not in the document]
Comment (unbreakabl3) [reply vro-types/o11n-plugin-activedirectory/index.d.ts:742]: Fixed

File: vro-types/o11n-plugin-activedirectory/index.d.ts [hunk 3]
```diff
@@ -490,21 +563,183 @@ declare interface AD_OrganizationalUnit {
...
+
+	/**
+	 * Get an AD attribute value as byte array.
+	 */
+	getAttributeValueBytes(attribName: string): object[];
+
+	/**
+	 * Get an AD attribute for an array of values.
+	 */
+	getAttributeValueBytes(attribName: string): string[];
```
[the remark this answers is not in the document]
Comment (unbreakabl3) [reply vro-types/o11n-plugin-activedirectory/index.d.ts:675]: Fixed

**Review by unbreakabl3 — COMMENTED** [review]
Fixed the issue after the code review

**Review by VenelinBakalov — APPROVED** [review]
Thanks!!

**Comment by VenelinBakalov** [comment]
I will merge it in order to not create you more merge conflicts :)

**Pull request merged**